services/network_config_manager/src/main/java/com/futurewei/alcor/netwconfigmanager/server/grpc/GoalStateProvisionerServer.java

```diff
@@ -105,6 +128,17 @@ public void onNext(Goalstate.GoalStateV2 value) {
                     Map<String, HostGoalState> hostGoalStates = NetworkConfigManagerUtil.splitClusterToHostGoalState(value);
 
                     //store the goal state in cache
+                    Set<String> processedResourceIds = new HashSet<>();
+                    for (Map.Entry<String, HostGoalState> entry : hostGoalStates.entrySet()) {
+                        String hostId = entry.getKey();
+                        HostGoalState hostGoalState = entry.getValue();
+
+                        try {
+                            goalStatePersistenceService.updateGoalState(hostId, hostGoalState);
+                        } catch (Exception e) {
+                            e.printStackTrace();
+                        }
+                    }
```

Comment:
TXN:
This entire loop can be pushed into goalStatePersistenceService.updateGoalState() and the call it self invoked after grpcGoalStateClient.sendGoalStates(filteredGoalStates); in the try{} block below.

The abouve loop inside goalStatePersistenceService.updateGoalState()  will have to be modified as follows:
goalStatePersistenceService.updateGoalState() {
try {
Transaction txn = getTransaction();
txn.start();
<The loop body over the current code>
txn.commit();
} cartch () {}

This way, if the send fails we don't have to do any rollbacks.

I have added
public Transaction getTransaction() {
        if(hrmCache != null)
            return hrmCache.getTransaction();
        if (rssCache != null)
            return rssCache.getTransaction();

        if (vpcCache != null)
            return vpcCache.getTransaction();

        return null;
    }

In my equivalent of service/impl/GoalStatePersistenceServiceImpl.java.

There is the issue of capturing changes from updates to ACA into NCM caches, which, I don't think is handled by this code. I might be wrong, in which case a clear comment to that effect will help a lot.

Reply:
Sounds good. As discussed, we will merge the current NCM implementation for non-transactional perf test. Meanwhile please work on the other PR on Transaction Mode changes (PR #604) and get it merged. 

Next step is continue refactoring this part to enable transactional support and do a quick perf test.